transaction/core/src/input_rules.rs

```diff
@@ -36,32 +36,33 @@ pub struct InputRules {
     #[prost(fixed64, tag = "2")]
     pub max_tombstone_block: u64,
 
-    // Outputs required to appear in the TxPrefix, but which are permitted to be
-    // filled partially instead of fully, according to the "fill fraction" which
-    // is inferred using the "partial fill change" output (MCIP #42)
+    /// Outputs required to appear in the TxPrefix, but which are permitted to
+    /// be filled partially instead of fully, according to the "fill
+    /// fraction" which is inferred using the "partial fill change" output
+    /// (MCIP #42)
     #[prost(message, repeated, tag = "3")]
     pub partial_fill_outputs: Vec<RevealedTxOut>,
 
-    // A change output for any leftover from this input, which may occur during
-    // a partial fill (MCIP #42).
-    //
-    // This field must be present whenever partial fills are used, because the
-    // comparison of this "idealized" output and the corresponding "fractional"
-    // change which appears in the TxPrefix is what determines the "fill fraction",
-    // that is, the degree to which a counterparty is obliged to fill every
-    // partial fill output in these rules.
-    //
-    // It is an error to use any of the partial fill options without also
-    // setting this.
+    /// A change output for any leftover from this input, which may occur during
```

Comment:
:thought_balloon: Thoughts on `partial_fill_input` as a counter to `partial_fill_outputs`

Reply:
my thinking is that, usually in our system an Input means:

* A specific UTXO that you are adding to a transaction
* An input ring (containing said UTXO and some mixins), which becomes a TxIn in the prefix

The `partial_fill_change` really has nothing to do with that UTXO (except that it might have the same value) -- it is a TxOut object, which is required to appear (possibly in a modified form) in the tx.prefix.outputs list.

Reply:
I was thinking about TxOuts the other day.  

Part of my hiccups have been, I see a function and internally ask what are the inputs and outputs.  Then I see `TxOut` and instinctively want to mark it an output. As you say, it's often an input in the code. It is named appropriately as it's a data structure which is an _output_ of the final transaction.

Reply:
Yeah I remember discussing this a lot with Kyle, it's a stumbling block for people that TxOut is an input to a transaction.

I don't know if it's actionable though -- the idea of the UTXO blockchain model is that every input to a transaction is the output of an earlier transaction, and the industry term for these objects is TXO / UTXO. If we depart too much from this terminology it's going to be confusing to developers from other blockchains, and to users who are familiar with other blockchains.

Reply:
Like ultimately, these terms come from Bitcoin

Reply:
I agree common industry terms should win out.  I think it's just something I'll need to keep in mind

Thanks!